packages/firestore/lite/src/api/reference.ts

```diff
@@ -318,6 +318,30 @@ export function collection(
   }
 }
 
+export function collectionGroup(
+  firestore: firestore.FirebaseFirestore,
+  collectionId: string
+): Query<firestore.DocumentData> {
+  const firestoreClient = cast(firestore, Firestore);
+
+  validateArgType('collectionGroup', 'non-empty string', 1, collectionId);
+  if (collectionId.indexOf('/') >= 0) {
+    throw new FirestoreError(
+      Code.INVALID_ARGUMENT,
+      `Invalid collection ID '${collectionId}' passed to function ` +
+        `collectionGroup(). Collection IDs must not contain '/'.`
+    );
+  }
```

Comment:
Suggestion: Factor into a separate assertion method.

Reply:
This is a special error that only applies to collection IDs. The `collection` and `doc` methods can be called with `/`-separated paths.